polars/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -217,25 +217,36 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
                 reg.replace(s, val)
             }
         };
+        let ca = self.as_utf8();
         Ok(ca.apply(f))
     }
 
     /// Replace the leftmost literal (sub)string with another string
-    fn replace_literal(&self, pat: &str, val: &str) -> PolarsResult<Utf8Chunked> {
-        self.replace(escape(pat).as_str(), val)
+    fn replace_literal<'a>(&'a self, pat: &str, val: &str) -> PolarsResult<Utf8Chunked> {
+        let reg = Regex::new(escape(pat).as_str())?;
+        let f = |s: &'a str| {
```

Comment:
Why do we actually use regex when we replace literal? :thinking:

Reply:
Speed; it's actually faster on longer strings - we use standard string replace on strings < 32 chars. See https://github.com/pola-rs/polars/pull/4057 for all the relevant graphs :)

Reply:
Ah, check! 

I think we should do a new benchmark in a year. Rust std might catch up with burntsushi on this.

Reply:
> I think we should do a new benchmark in a year. Rust std might catch up with burntsushi on this.

Lol... I found the script (I'll gist it for you if you like) and am running it now to se if we can simplify on one or the other ;)